### PR TITLE
CTP-1582, issue #43 : Fixed all GHA issues.

### PR DESCRIPTION
**Pull request by opitz**
removed IE specific gradient fallback in CSS
removed duplicate background declarations in style.css. re-established image variable in mustache template fixed issue 4: wrong position of optional parameter test grunt against MOODLE_4x only due to staleness issue with MOODLE_311_STABLE

issue #42 : replaced strings for link to Student Record System and text on related launch button with site level plugin settings.
issue #41 : re-defined archive links to Academic Year dropdown
issue #44 : fixed order for required parameter

### CODE REVIEW

**Comment by aspark21** [comment]
@opitz before I start the peer review, there's a merge conflict - can you review & resolve / re-base on the latest main branch

**Comment by opitz** [comment]
@aspark21 done!

File: templates/progress.mustache [hunk 1]
```diff
@@ -17,16 +17,20 @@
...
-    This template renders the progress spinner.
+    This template renders a progress GIF.
+
+    Variables required for this template:
+    * progressimg - The URL to the progress image
 
     Example context (json):
     {
-        "progressimg": "/report/myfeedback/pix/progress.gif",
+        "progressimg": "https://localhost/report/myfeedback/pix/progress.gif"
```
Comment (aspark21) [inline templates/progress.mustache]: are you sure this has to be absolute path rather than relative

File: student/feedback.php [hunk 2]
```diff
@@ -14,10 +14,11 @@
 // You should have received a copy of the GNU General Public License
 // along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
 
-/*
+/**
```
Comment (aspark21) [inline student/feedback.php:17]: /**
but
*/
doesn't seem consistent

is that right?

**Comment by aspark21** [comment]
have you deployed this on a preview instance I can take a look at? - especially for #41 and #42

**Comment by aspark21** [comment]
2 minor questions but review done, otherwise fine

and noted the removal of a lot of !important for css

File: student/feedback.php (hunk 2 above)
Comment (opitz) [reply student/feedback.php:17]: Yes, that is actually right. (See any other file: this comment block starts with "/**)

File: templates/progress.mustache (hunk 1 above)
Comment (opitz) [reply templates/progress.mustache]: Thank you for this. Obviously this is a flaw...

**Comment by opitz** [comment]
@aspark21 I have deployed to https://410-codebase.preview-moodle.ucl.ac.uk/

File: student/feedback.php (hunk 2 above)
Comment (aspark21) [reply student/feedback.php:17]: I guess I meant should it close with **/ to match but yeah I can see core is asymmetrical too

**Comment by opitz** [comment]
@aspark21 Does this need any further attention or can it be merged now?

**Comment by aspark21** [comment]
Creating mega merge requests with many different things in one is a sure fire way to get this stalled. 

Can you restrict this change to coding standard fixes - easy to review and approve - and split off changes relating to other issues into their own merge requests please.

**Comment by aspark21** [comment]
#41 in particular

**Comment by aspark21** [comment]
@opitz Feedback on #41 which has been holding this up:
Academic year setting and making past academic years computed based on that doesn't make sense / work for us. And it means having to update a bunch of these settings every academic year, which will most likely be overlooked.

Can we just have a label field for each to describe the academic year they relate to.


Wider piece:
Can you revert [0706045](https://github.com/uclmoodle/moodle-report_myfeedback/pull/45/commits/07060456747bbfecdebb71d15dbe19422e0a0f0f) or re-re-grunt, MDL-75012 was reverted in https://tracker.moodle.org/browse/MDL-77538 so this will have changed again

**Comment by aspark21** [comment]
replaced by #47

**Pull request closed without merging**